Main.py

```diff
@@ -553,6 +553,19 @@ def find_light_arrows(spoiler):
         maybe_set_light_arrows(location)
 
 
+def maybe_set_song_of_time(location):
```

Comment:
Every change I made in this file is lifted and shifted from light_arrow_location logic. I didn't take the time to understand what this is doing.